cosmwasm/packages/sgx-vm/src/enclave.rs

```diff
@@ -68,16 +102,20 @@ fn init_enclave() -> SgxResult<SgxEnclave> {
 
 #[allow(clippy::mutex_atomic)]
 lazy_static! {
-    static ref SGX_ENCLAVE: SgxResult<SgxEnclave> = init_enclave();
+    static ref SGX_ENCLAVE_MUTEX: SgxResult<EnclaveMutex> = EnclaveMutex::new();
+
     /// This variable indicates if the enclave configuration has already been set
     static ref SGX_ENCLAVE_CONFIGURED: Mutex<bool> = Mutex::new(false);
 }
 
 /// Use this method when trying to get access to the enclave.
 /// You can unwrap the result when you are certain that the enclave
 /// must have been initialized if you even reached that point in the code.
-pub fn get_enclave() -> SgxResult<&'static SgxEnclave> {
-    SGX_ENCLAVE.as_ref().map_err(|status| *status)
+/// If `Ok(None)` is returned, that means that the enclave is currently busy.
+pub fn get_enclave() -> SgxResult<Option<EnclaveGuard>> {
+    let mutex = SGX_ENCLAVE_MUTEX.as_ref().map_err(|status| *status)?;
+    let maybe_guard = mutex.get_enclave(Duration::from_secs(6));
```

Comment:
Make this a const instead of a magic number, even if it isn't fully customizable

Reply:
done 👍🏻